lib/ansible/modules/system/syspatch.py

```diff
@@ -114,6 +114,9 @@ def syspatch_run(module):
     elif module.params['apply']:
         check_flag = ['-c']
         run_flag = []
+    else:
```

Comment:
`apply` parameter at line 85 is set to 'False' so `required_one_of` at line 92 is not executed correctly. You need to remove `default=False` from line 85 to make this work rather than adding else statement here.

Also, 
1. On line 101, change the current line to `cmd = module.get_bin_path('syspatch', True)` so it will check the find the `syspatch` binary for the user from user executable path.

2. On line 144 and line 148, change line `warnings.append[` to `warnings.append(` since `warnings` is a list.

Reply:
Add changelog entry for this change [here](https://github.com/ansible/ansible/tree/devel/changelogs/fragments)

Reply:
Thank you for commenting, @Akasurde 

How will these changes ensure `check_flag` is set when it's referenced in line 122? I did not find the definition for `required_one_of()`, maybe it would set `check_flag`?

> You need to remove default=False from line 85 to make this work rather than adding else statement here.

[The doc](https://docs.ansible.com/ansible/latest/modules/syspatch_module.html#syspatch-module) says `apply` will default to no/False. If we remove `default=False` here, where will the default be applied?

The line 101 and 144,148 changes you proposed feel out of scope for this pull request, I may just misunderstand tho. Will those changes help handle unset check and run flags?

Reply:
[`required_one_of`](https://github.com/ansible/ansible/blob/63b94fca7e96a0fe9fcb92d3af3097e3a309f5ab/lib/ansible/module_utils/basic.py#L583) is used to check either `revert` or `apply` is provided. On line 85 `apply` is set to `default=False` so causes [`check_required_one_of`](https://github.com/ansible/ansible/blob/63b94fca7e96a0fe9fcb92d3af3097e3a309f5ab/lib/ansible/module_utils/common/validation.py#L71) to fail check this condition and goes till `syspatch_run` method and fails to set `check_flag` correctly.

Yes, 101 and 144,148 changes are not related to this change and can be omittd from this patch. But since you are here, you can fix them.

Reply:
Thank you for explaining 🙇